### PR TITLE
[WIP] Support for np.trim_zeros

### PR DESCRIPTION
**Pull request by rjenc29**
Initial commit for CI and to discuss failing test case.

### CODE REVIEW

**Comment by rjenc29** [comment]
I thought I would try implementing `np.trim_zeros` as it has a couple of interesting features:
1. One of the arguments is a string
2. The return type is supposed to be the same as the provided input (i.e. list in, list out etc etc)

On the former, the 2.7 builds have all errored but the others look OK.  

On the latter, this works as expected for array and list (i.e. you get the same data structure out as you originally passed in).  But the approach doesn't work for tuples, as it would mean slicing with indices which aren't known up front.

I'd appreciate your thoughts on these points (apologies if one or both is obvious!) when you get a mo.

**Comment by stuartarchibald** [comment]
Thanks for the PR, this needs Py2 basic string (`bytes`) support so am marking it as blocked awaiting long term feature. Numba core developer meeting reference: https://github.com/numba/numba/wiki/Minutes_2018_12_11

**Comment by rjenc29** [comment]
Great, thanks for the update and I will keep an eye out.

**Comment by stuartarchibald** [comment]
Python 2.7 is now out of Numba's support remit, so this PR is unblocked!

File: numba/targets/arraymath.py [hunk 1]
```diff
@@ -1294,6 +1294,60 @@ def np_vander_seq_impl(x, N=None, increasing=False):
...
+        DEFAULT_TRIM = True
+
+    def np_trim_zeros_impl(filt, trim='fb'):
+        filt_arr = _asarray(filt)
+
+        if len(filt_arr) == 0:
+            return filt
+
+        if DEFAULT_TRIM:
+            trim = 'fb'
```
Comment (eric-wieser) [inline numba/targets/arraymath.py]: Is this necessary? Won't this already be set this way anyway?

**Pull request closed without merging**